reactor/http_action.go

```diff
@@ -36,25 +36,41 @@ func (hta *HTTPAction) getURL() string {
 	}
 	jkPort := hta.auth.Port
 
-	url, ok := hta.actionInfo.Params["url"]
+	url, ok := hta.actionInfo.Params["query"].(map[string]interface{})["url"]
 	if !ok {
 		hta.GetLogger().Errorln("URL is not defined to the HTTP action. Check your configuration.")
 		return ""
 	}
 
-	out := url.(string)
-	if !strings.HasPrefix(out, "http") && strings.Contains(out, "://") {
-		var buff strings.Builder
+	var buff strings.Builder
+
+	addr := url.(string)
+
+	if strings.Contains(addr, "://") {
+		buff.WriteString(addr)
+	} else {
 		buff.WriteString("https://")
 		buff.WriteString(jkFqdn)
 		if jkPort != 443 && jkPort != 0 {
 			buff.WriteString(":" + strconv.Itoa(jkPort))
 		}
-		buff.WriteString("/" + strings.TrimPrefix(path.Clean(out), "/"))
-		out = buff.String()
+		buff.WriteString("/" + strings.TrimPrefix(path.Clean(addr), "/"))
+	}
+
+	query := hta.actionInfo.Params["query"].(map[string]interface{})["params"]
+	first := true
+	for qkey, qval := range query.(map[string]interface{}) {
+		var sep string
+		if first {
+			sep = "?"
+			first = false
+		} else {
+			sep = "&"
+		}
+		buff.WriteString(sep + qkey + "=" + qval.(string))
```

Comment:
The purpose of this function is basically to add basic hostname or call something else, in case the entire query also contains full URL. So that means, if there is `/something/like/this` then it should be converted to `$SCHEME://$JENKINS_HOST/something/like/this`. If there is full URL, like `https://myown.jenkins/something/like/this` then it is left untouched.

Another thing is that each URL has also `params` section, because it still doesn't mean that everything goes GET method. That said, `?foo=bar` should be actually described in `params` section like so:

```yaml
query:
  url: /something/like/that
  method: post # or get
  params:
    foo: bar
```

Then all that needs to be constructed accordingly.

Reply:
Actually this is POST (GET is not implemented) but yes oddly this worked.
In POST these should indeed not be in URI...